### PR TITLE
Adding a configuration for parsing command-line args in shipping builds 

### PR DESCRIPTION
**Pull request by improbable-danny**
#### Description
This PR adds a configuration for our internal shipping builds to use checks, logs and additionally parse command-line arguments.

Engine PR: https://github.com/improbableio/UnrealEngine/pull/573

#### Tests
Using with the NFR pipeline

STRONGLY SUGGESTED: How can this be verified by QA?
Try to use spatial cmd line args in shipping (OverrideSpatialNetworking) with and without this new target rule setting.

#### Primary reviewers
@m-samiec

### CODE REVIEW

**Comment by MatthewSandfordImprobable** [comment]
I'm interested in what purpose this serves?

**Comment by improbable-danny** [comment]
> I'm interested in what purpose this serves?

It allows us to disable spatial in shipping builds for native tests, there is some context in the engine PR https://github.com/improbableio/UnrealEngine/pull/573/files#diff-037dc1bb06f6c1a03fbaeb50039e3dc0R105

File: SpatialGDK/Source/SpatialGDK/Private/SpatialGDKSettings.cpp [hunk 1]
```diff
@@ -36,11 +37,13 @@ namespace
 				bOutValue = FCString::ToBool(TempStr + 1); // + 1 to skip =
 			}
 		}
+#endif
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/SpatialGDKSettings.cpp:40]:
```suggestion
#endif // ALLOW_SPATIAL_CMDLINE_PARSING
```

File: SpatialGDK/Source/SpatialGDK/Private/SpatialGDKSettings.cpp [hunk 2]
```diff
@@ -53,6 +56,7 @@ namespace
...
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/SpatialGDKSettings.cpp:59]:
```suggestion
#endif // ALLOW_SPATIAL_CMDLINE_PARSING
```

**Pull request merged**